src/modules/bulletproofs/rangeproof_uncompressed_impl.h

```diff
@@ -143,4 +143,135 @@ static int secp256k1_bulletproofs_rangeproof_uncompressed_prove_step0_impl(
     return 1;
 }
 
+/* Step 1 of the proof.
+ * Uses the value but not the blinding factor. Takes the complete commitment,
+ * but only to put it into the hash.
+ * Outputs the points T1 and T2, encoded in 65 bytes (one byte with T1's parity
+ * in the LSB, T2's parity in the second-LSB, 32-byte T1.x, 32-byte T2.x).
+ * Updates the state to include the hash x.
+ */
+static int secp256k1_bulletproofs_rangeproof_uncompressed_prove_step1_impl(
+    const secp256k1_ecmult_gen_context* ecmult_gen_ctx,
+    secp256k1_bulletproofs_prover_context* prover_ctx,
+    unsigned char* output,
+    const size_t n_bits,
+    const uint64_t value,
+    const uint64_t min_value,
+    const secp256k1_ge* asset_genp,
+    const unsigned char* nonce
+) {
+    secp256k1_bulletproofs_bulletproofs_lrgen lr_gen;
+    secp256k1_sha256 sha256;
+    secp256k1_scalar tau1, tau2;
+    secp256k1_scalar t0, t1, t2;
+    secp256k1_scalar y, z;
+    secp256k1_scalar tmps;
+    secp256k1_gej gej;
+    secp256k1_ge ge;
+    size_t i;
+    int overflow;
+
+    memset(output, 0, 65);
+
+    /* Unpack challenges y and z from step 0 */
+    secp256k1_scalar_set_b32(&y, &prover_ctx->data[32], &overflow);
+    if (overflow || secp256k1_scalar_is_zero(&y)) {
+        memset(prover_ctx->data, 0, sizeof(prover_ctx->data));
+        return 0;
+    }
+    secp256k1_scalar_set_b32(&z, &prover_ctx->data[64], &overflow);
+    if (overflow || secp256k1_scalar_is_zero(&z)) {
+        memset(prover_ctx->data, 0, sizeof(prover_ctx->data));
+        memset(output, 0, 65);
+        return 0;
+    }
+
+    /* Compute coefficients t0, t1, t2 of the <l, r> polynomial */
+    /* t0 = l(0) dot r(0) */
```

Comment:
We should definitely stick with `dot` if we're gonna do anything, since this is easier to search for and easier to put into variable names (e.g. `l_dot_r`). I'll change `<l, r>` to `(l dot r)`.